runtime/cmake/J9vmTargetSupport.cmake

```diff
@@ -20,15 +20,50 @@
 # SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
 ################################################################################
 
+if(J9VM_TARGET_SUPPORT)
+	return()
+endif()
+set(J9VM_TARGET_SUPPORT 1)
+
 include(OmrTargetSupport)
 
-# Currently j9vm_add_library and j9vm_add_executable just call
-# omr_add_library and omr_add_executable, but have them so that we can add
-# functionality in the future, without having to hunt down all the 
-# add_library/add_executable calls.
+# Currently j9vm_add_library and j9vm_add_executable are wrappers arroung
+# omr_add_library and omr_add_executable. Here we only document the additional
+# functionality provided. For full information see cmake/moudles/OmrTargetSupport.cmake
+# from omr
+
+# j9vm_add_library(<target_name> <library_type> [options ...] [source ...] )
+#  NO_VERSION_SUFFIX  - Don't append a version suffix  (eg 29) on the output file name.
+#                       Note: Only applies to SHARED libraries
+function(j9vm_add_library name type)
+	set(options NO_VERSION_SUFFIX)
+	set(oneValueArgs OUTPUT_NAME)
+	set(multiValueArgs)
+	cmake_parse_arguments(opt "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+	# Check if we need to append a version suffix eg '29'
+	set(output_name_args)
+	if(type STREQUAL "SHARED")
+		# If OUTPUT_NAME is specified, use that as the starting name
+		# otherwise the name of the cmake target is our starting point
+		if(opt_OUTPUT_NAME)
+			set(output_name "${opt_OUTPUT_NAME}")
+		else()
+			set(output_name "${name}")
+		endif()
+
+		# Default to appending a version suffix, unless we are told not to
+		if(NOT opt_NO_VERSION_SUFFIX)
```

Comment:
If `OUTPUT_NAME` is explicitly specified, I don't think the version suffix should be added even if `NO_VERSION_SUFFIX` was not specified.